src/fetchtastic/downloader.py

```
@@ -639,46 +650,234 @@ def get_file_hash(file_path: str) -> Optional[str]:
 
     return hash1 == hash2
 
+    try:
+        with open(text_file, "r", encoding="utf-8") as f:
```

Ruff (0.14.3)

Undefined name `text_file`

(F821)